apps/client/src/app/core/scheme/categories/categories.component.ts

```diff
@@ -0,0 +1,181 @@
+  constructor(
+    @Inject(Toast) private toast: Toast,
+    private confirmationService: ConfirmationService,
+    private categoryService: CategoryService
+  ) {
+    const initialControlValues = {
+      name: '',
+      helpText: ''
+    }
+    const validationSchema = Yup.object({
+      name: Yup.string()
+        .required('El nombre es requerido')
+        .max(50, 'El nombre no puede superar los 50 caracteres'),
+      helpText: Yup.string().required('La ayuda es requerida')
+    })
+
+    super(initialControlValues, validationSchema)
+  }
+
+  errors = {
+    name: '',
+    helpText: ''
+  }
+  @Input() categories: Category[] = []
+  @Input() indicatorIndex: number = 0
+  visibleCreate = false
+  visibleEdit = false
+  fetchCreateCategory: boolean = false
+  fetchEditCategory: boolean = false
+  categoryEdit: formPayloadEdit = {
+    name: '',
+    helpText: '',
+    id: 0
+  }
+  closeDialog() {
+    this.visibleCreate = false
+    this.visibleEdit = false
+    this.formGroup.reset()
+    this.errors = {
+      name: '',
+      helpText: ''
+    }
+  }
+
+  showDialogEdit(category: Category) {
+    this.visibleEdit = true
+    this.categoryEdit = category
+    this.formGroup.controls.name.setValue(this.categoryEdit.name)
+    this.formGroup.controls.helpText.setValue(this.categoryEdit.helpText)
+  }
+  confirmationDelete(event: Event, id: number, name: string) {
+    this.confirmationService.confirm({
+      target: event.target as EventTarget,
+      message: `¿Estás seguro de que quieres eliminar la categoría <strong>${name}</strong>?`,
+      header: 'Eliminar categoría',
+      icon: 'pi pi-info-circle',
+      acceptButtonStyleClass: 'p-button-danger p-button-text',
+      rejectButtonStyleClass: 'p-button-text p-button-text',
+      acceptIcon: 'none',
+      rejectIcon: 'none',
+      acceptLabel: 'Sí',
+      rejectLabel: 'No',
+
+      accept: () => {
+        this.toast.show('info', 'Eliminando..', 'Eliminando criterio..')
+        this.onDelete(id)
+      },
+      reject: () => {
+        this.toast.show('error', 'Rechazado', 'Haz rechazado la eliminación')
+      }
+    })
+  }
+  getCategoriesPerIndicator(indicatorId: number) {
+    this.categoryService
+      .getAllCategoriesByIndicatorIndex(indicatorId)
+      .subscribe({
+        next: (res) => {
+          this.categories =
+            res.status === 'success' ? res.data : this.categories
+        },
+        error: (e) => {
+          console.error(e)
+          this.toast.show('error', 'Error', e.error.data.message)
+        }
+      })
+  }
+  onDelete(id: number) {
+    this.categoryService.delete(id).subscribe({
+      next: () => {
+        this.toast.show('success', 'Eliminado', 'Categoría eliminada con éxito')
+        this.getCategoriesPerIndicator(this.indicatorIndex)
+      },
+      error: (e) => {
+        console.error(e)
+        this.toast.show('error', 'Error', e.error.data.message)
+      }
+    })
+  }
+  onCreate() {
```

Comment:
No estás chequeando que el form sea válido

```suggestion
  onCreate() {
    if (this.formGroup.invalid) return
    ...
```